### ISSUE TITLE
OVH plugin tests require internet access

### ISSUE THREAD

**Issue by hlieberman** [certbot/certbot#6222]
Unlike other plugins using the lexicon backend, the dns-ovh plugin attempts to hit the actual OVH API during the testing phase.  When this is attempted inside Debian build environments, the tests all fail as internet access is not available.

```
test_add_txt_record (certbot_dns_ovh.dns_ovh_test.OVHLexiconClientTest) ... ERROR
test_add_txt_record_error_adding_record (certbot_dns_ovh.dns_ovh_test.OVHLexiconClientTest) ... ERROR
test_add_txt_record_error_finding_domain (certbot_dns_ovh.dns_ovh_test.OVHLexiconClientTest) ... ERROR
test_add_txt_record_fail_to_authenticate (certbot_dns_ovh.dns_ovh_test.OVHLexiconClientTest) ... ERROR
test_add_txt_record_fail_to_authenticate_with_unknown_error (certbot_dns_ovh.dns_ovh_test.OVHLexiconClientTest) ... ERROR
test_add_txt_record_fail_to_find_domain (certbot_dns_ovh.dns_ovh_test.OVHLexiconClientTest) ... ERROR
test_add_txt_record_try_twice_to_find_domain (certbot_dns_ovh.dns_ovh_test.OVHLexiconClientTest) ... ERROR
test_del_txt_record (certbot_dns_ovh.dns_ovh_test.OVHLexiconClientTest) ... ERROR
test_del_txt_record_error_deleting_record (certbot_dns_ovh.dns_ovh_test.OVHLexiconClientTest) ... ERROR
test_del_txt_record_error_finding_domain (certbot_dns_ovh.dns_ovh_test.OVHLexiconClientTest) ... ERROR
test_del_txt_record_fail_to_authenticate (certbot_dns_ovh.dns_ovh_test.OVHLexiconClientTest) ... ERROR
test_del_txt_record_fail_to_authenticate_with_unknown_error (certbot_dns_ovh.dns_ovh_test.OVHLexiconClientTest) ... ERROR
test_del_txt_record_fail_to_find_domain (certbot_dns_ovh.dns_ovh_test.OVHLexiconClientTest) ... ERROR
```
```
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='eu.api.ovh.com', port=443): Max retries exceeded with url: /1.0/auth/time (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f8c1e181ac8>: Failed to establish a new connection: [Errno 111] Connection refused',)))
```

**Comment by elyscape** [comment]
This appears to be caused by the OVH provider's initializer [making an un-mocked request to the time endpoint][init]. This seems like something that should be moved into upstream's authenticate method.

[init]: https://github.com/AnalogJ/lexicon/blob/d697ed1caa1942105c746bddadc12ed68181807f/lexicon/providers/ovh.py#L51-L56

**Comment by adferrand** [comment]
I developed the OVH provider for Lexicon. It was my first provider, so I did some design error. 

@elyscape Do you mean that if the session is opened during the authenticate method call, it will be correctly mocked by certbot unit tests? 

I will make the PR on lexicon and update the issue when it is merged to make certbot maintainers update lexicon version dependency.

**Comment by ohemorange** [comment]
Closed in #6380
